core/src/main/java/org/apache/iceberg/rest/RESTSerializers.java

```diff
@@ -91,8 +94,12 @@ public static void registerAll(ObjectMapper mapper) {
         .addDeserializer(
             ImmutableReportMetricsRequest.class, new ReportMetricsRequestDeserializer<>())
         .addSerializer(CommitTransactionRequest.class, new CommitTransactionRequestSerializer())
+        .addDeserializer(CommitTransactionRequest.class, new CommitTransactionRequestDeserializer())
+        .addSerializer(RegisterTableRequest.class, new RegisterTableRequestSerializer<>())
+        .addDeserializer(RegisterTableRequest.class, new RegisterTableRequestDeserializer<>())
```

Comment:
Why do we register both the interface and the implementation? Shouldn't we just need one?

Reply:
Removing any one of them throws the below response. I see `ReportMetricsRequest` register both the interface and the implementation.

```
    org.apache.iceberg.exceptions.ServiceFailureException: Server error: null: {
        "servlet":"org.apache.iceberg.rest.RESTCatalogServlet-4c38cd16",
        "cause0":"com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.apache.iceberg.rest.requests.RegisterTableRequest` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information\n at [Source: (org.eclipse.jetty.server.Request$1); line: 1, column: 1]",
        "message":"com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.apache.iceberg.rest.requests.RegisterTableRequest` (no Creators, like default constructor, exist): abstract types either need to be mapped to concrete types, have custom deserializer, or contain additional type information\n at [Source: (org.eclipse.jetty.server.Request$1); line: 1, column: 1]",
        "url":"/v1/namespaces/newdb/register",
        "status":"500"
        }
```

Reply:
we need to register both classes unfortunately. See also https://github.com/apache/iceberg/pull/5407#discussion_r984472975 for some context and  https://github.com/apache/iceberg/pull/5911 where I attempted to fix it